homeassistant/components/flo/services.py

```diff
@@ -0,0 +1,109 @@
+"""Services for the Flow by Moen integration."""
+from aioflo.api import API
+from aioflo.location import SLEEP_MINUTE_OPTIONS, SYSTEM_MODE_HOME, SYSTEM_REVERT_MODES
+import voluptuous as vol
+
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import CLIENT, DOMAIN as FLO_DOMAIN
+
+ATTR_LOCATION_ID = "location_id"
+ATTR_DEVICE_ID = "device_id"
+ATTR_REVERT_TO_MODE = "revert_to_mode"
+ATTR_SLEEP_MINUTES = "sleep_minutes"
+
+DEVICE_ID_SERVICE = "device_id_service"
+LOCATION_ID_SERVICE = "location_id_service"
+
+SERVICE_SET_SLEEP_MODE = "set_sleep_mode"
+SERVICE_SET_AWAY_MODE = "set_away_mode"
+SERVICE_SET_HOME_MODE = "set_home_mode"
+SERVICE_RUN_HEALTH_TEST = "run_health_test"
+
+SERVICE_SCHEMAS = {
+    DEVICE_ID_SERVICE: vol.Schema(
+        {vol.Required(ATTR_DEVICE_ID, default=None): cv.string}
+    ),
+    LOCATION_ID_SERVICE: vol.Schema(
+        {vol.Required(ATTR_LOCATION_ID, default=None): cv.string}
+    ),
+    SERVICE_SET_SLEEP_MODE: vol.Schema(
+        {
+            vol.Required(ATTR_LOCATION_ID, default=None): cv.string,
+            vol.Required(ATTR_SLEEP_MINUTES, default=120): vol.In(SLEEP_MINUTE_OPTIONS),
+            vol.Required(ATTR_REVERT_TO_MODE, default=SYSTEM_MODE_HOME): vol.In(
+                SYSTEM_REVERT_MODES
+            ),
+        }
+    ),
+}
+
+
+@callback
+def async_load_services(hass: HomeAssistantType):
+    """Load the services exposed by the Flo component."""
+
+    async def async_set_mode_home(service):
+        """Set the Flo location to home mode."""
+        location_id: str = service.data.get(ATTR_LOCATION_ID)
```

Comment:
Preferably you would allow targeting entities for this service call, so that the user doesn't need to know about location IDs (since you can lookup entity -> location in code). To do so, we would register it as an entity service to one of the Flo platforms. Not sure if there is one "main" platform (ie for sonos it's media player)

Reply:
After looking at the switch integration, that's the main one. So let's register it on the switch platform.

Reply:
yeah that definitely makes the most sense here. I'll finish this PR up after I merge switch so I can rebase this and continue.

Reply:
done